python/SAGEMAKER_NOTEBOOK_NO_DIRECT_INTERNET_ACCESS/SAGEMAKER_NOTEBOOK_NO_DIRECT_INTERNET_ACCESS.py

```diff
@@ -0,0 +1,410 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+    SAGEMAKER_NOTEBOOK_NO_DIRECT_INTERNET_ACCESS
+
+Description:
+  Check whether direct internet access is disabled for an Amazon SageMaker Notebook Instance.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::::Account
+
+Scenarios:
+  Scenario: 1
+     Given: No Amazon SageMaker notebook instances exist
+      Then: Return NOT_APPLICABLE
+  Scenario: 2
+     Given: At least one Amazon SageMaker notebook instance exists
+       And: DirectInternetAccess is set to Enabled for the Amazon SageMaker notebook instance
+      Then: Return NON_COMPLIANT with annotation "This Amazon SageMaker Notebook Instance has direct internet access."
+  Scenario: 3
+     Given: At least one Amazon SageMaker notebook instance exists
+       And: DirectInternetAccess is set to Disabled for the Amazon SageMaker notebook instance
+      Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    evaluations = []
+
+    sagemaker_client = get_client('sagemaker', event)
+    notebook_instance_list = get_all_notebook_instances(sagemaker_client)
+
+    #SCENARIO 1: No Amazon SageMaker notebook instances exist
+    if not notebook_instance_list:
+        return None
```

Comment:
Return NOT_APPLICABLE

Reply:
@shikharj05 On returning None, the RDK automatically handles it and returns NOT_APPLICABLE. I'm not sure if it should be returned explicitly as "NOT_APPLICABLE"
@jongogogo any comments on this?

Reply:
`None` is OK